fog/overseer/server/tests/retired_key_not_outstanding_idles_have_same_key.rs

```diff
@@ -27,151 +17,75 @@ const PORT_NUMBER: u16 = 8084;
-        peer_indices.iter().cloned(),
-        recovery_db.clone(),
-        &watcher_path,
-        &ledger_db_path,
-        logger.clone(),
-    );
-    let (node1, _node1dir, client_listen_uri1) = utils::make_node(
-        1,
-        peer_indices.iter().cloned(),
-        recovery_db.clone(),
-        &watcher_path,
-        &ledger_db_path,
-        logger.clone(),
-    );
-    let (node2, _node2dir, client_listen_uri2) = utils::make_node(
-        2,
-        peer_indices.iter().cloned(),
-        recovery_db.clone(),
-        &watcher_path,
-        &ledger_db_path,
-        logger.clone(),
-    );
-
-    assert!(!node0.is_active());
-    assert!(!node1.is_active());
-    assert!(!node2.is_active());
-
+    let mut ctx = utils::TestContext::new(BASE_PORT, true, logger.clone());
+    let nodes = ctx.make_nodes(3);
     // Give RPC etc. time to start
-    std::thread::sleep(Duration::from_millis(1000));
+    sleep(Duration::from_secs(1));
 
-    node0.activate().expect("node0 failed to activate");
+    nodes[0].activate().expect("first node failed to activate");
 
-    assert!(node0.is_active());
-    assert!(!node1.is_active());
-    assert!(!node2.is_active());
+    assert!(nodes[0].is_active());
+    assert!(!nodes[1].is_active());
+    assert!(!nodes[2].is_active());
 
     // Get the original keys using node.
-    let original_node0_status = node0.get_ingest_summary();
-    let original_node0_ingress_key = original_node0_status.get_ingress_pubkey();
-    let original_node1_status = node1.get_ingest_summary();
-    let original_node1_ingress_key = original_node1_status.get_ingress_pubkey();
-    let original_node2_status = node2.get_ingest_summary();
-    let original_node2_ingress_key = original_node2_status.get_ingress_pubkey();
+    let original_node0_ingress_key = nodes[0].get_ingest_summary().get_ingress_pubkey().clone();
+    let original_node1_ingress_key = nodes[1].get_ingest_summary().get_ingress_pubkey().clone();
+    let original_node2_ingress_key = nodes[2].get_ingest_summary().get_ingress_pubkey().clone();
 
     assert_eq!(original_node0_ingress_key, original_node1_ingress_key);
     assert_eq!(original_node0_ingress_key, original_node2_ingress_key);
 
-    // Consider testing the CLI here instead. Initialize an OverSeerService
-    // object...
-    let rocket_config = rocket::Config::figment()
-        .merge(("port", PORT_NUMBER))
-        .merge(("address", "127.0.0.1"));
-    let mut overseer_service = OverseerService::new(
-        vec![client_listen_uri0, client_listen_uri1, client_listen_uri2],
-        recovery_db.clone(),
-        logger,
-    );
-    overseer_service.start().unwrap();
-    let overseer_state = OverseerState { overseer_service };
-    let rocket = server::initialize_rocket_server(rocket_config, overseer_state);
-    let client = Client::tracked(rocket).expect("valid rocket instance");
-    let _req = client.post("/enable").dispatch();
+    // TODO: Consider testing the CLI here instead.
```

Comment:
nit should todo go awy since it lives inside the `make_overseer_client_for_nodes()`
```suggestion
```